lib/elixir/lib/string_io.ex

```diff
@@ -263,35 +255,28 @@ defmodule StringIO do
 
   ## get_line
 
-  defp get_line(encoding, prompt,
-                %{input: input, output: output, capture_prompt: capture_prompt} = s) do
-    case :unicode.characters_to_list(input, encoding) do
-      {:error, _, _} ->
-        {{:error, :collect_line}, s}
-      {:incomplete, _, _} ->
-        {{:error, :collect_line}, s}
-      chars ->
-        {result, input} = do_get_line(chars, encoding)
-
-        s =
-          if capture_prompt do
-            %{s | output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
-          else
-            s
-          end
+  defp get_line(encoding, prompt, %{input: input} = s) do
+    with {:ok, count} <- bytes_until_eol(input, encoding, 0) do
+      {result, remainder} = case count do
+        0 -> {:eof, ""}
+        count -> split_at_eol(input, count)
+      end
 
-        {result, %{s | input: input}}
+      {result, state_after_read(s, remainder, prompt)}
+    else
+      _ -> {{:error, :collect_line}, s}
```

Comment:
I think this should match on `:error` to be explicit since this should be the only possible else clause, and could hide a bug.